# Laboratory excercise 5

The notebook contains excercise connected to the auditory excerice 6. For any questions feel free to contact assistant: eda.jovicic@fer.hr

In [4]:
import pandas as pd
import numpy as np
import statsmodels.formula.api as smf

# set seed for consistency
np.random.seed(2)

1. Load the cleaned dataset from the first exercise. If you haven't saved the dataset, rerun the exercise and save the final dataset.

In [12]:
#load data
df = pd.read_csv('students_cleaned_kopija.csv')
df

Unnamed: 0  Gender  EthnicGroup  ParentEduc     LunchType  TestPrep  \
0               0  female            2           1      standard         1   
1               1  female            2           5      standard         1   
2               2  female            1           3      standard         1   
3               3    male            0           0  free/reduced         1   
4               4    male            2           5      standard         1   
...           ...     ...          ...         ...           ...       ...   
29112       29112  female            3           2      standard         1   
29113       29113    male            4           2      standard         1   
29114       29114  female            2           2  free/reduced         0   
29115       29115  female            3           0      standard         0   
29116       29116    male            1           5      standard         1   

       ParentMaritalStatus  PracticeSport  IsFirstChild  NrSiblings  \
0                        1              1             1         3.0   
1                        1              2             1         0.0   
2                        2              2             1         4.0   
3                        1              0             0         1.0   
4                        1              2             1         0.0   
...                    ...            ...           ...         ...   
29112                    2              2             0         2.0   
29113                    2              1             0         1.0   
29114                    1              2             0         1.0   
29115                    1              1             0         3.0   
29116                    1              0             0         1.0   

       TransportMeans  WklyStudyHours  MathScore  ReadingScore  WritingScore  \
0                   1               1         71            71            74   
1                   2               0         69            90            88   
2                   1               1         87            93            91   
3                   2               0         45            56            42   
4                   1               0         76            78            75   
...               ...             ...        ...           ...           ...   
29112               1               0         59            61            65   
29113               0               0         58            53            51   
29114               0               0         61            70            67   
29115               1               0         82            90            93   
29116               1               0         64            60            58   

       Gender_female  Gender_male  LunchType_free/reduced  LunchType_standard  
0                1.0          0.0                     0.0                 1.0  
1                1.0          0.0                     0.0                 1.0  
2                1.0          0.0                     0.0                 1.0  
3                0.0          1.0                     1.0                 0.0  
4                0.0          1.0                     0.0                 1.0  
...              ...          ...                     ...                 ...  
29112            1.0          0.0                     0.0                 1.0  
29113            0.0          1.0                     0.0                 1.0  
29114            1.0          0.0                     1.0                 0.0  
29115            1.0          0.0                     0.0                 1.0  
29116            0.0          1.0                     0.0                 1.0  

[29117 rows x 19 columns]

2. Using linear regression, model the MathScore using features: TestPrep, NrSiblings, ParentMaritalStatus and ReadingScore. 

In [16]:
# Declares the model (MathScore as dependent variable, TestPrep, NrSiblings, ParentMaritalStatus and ReadingScore as independent)
mod = smf.ols(formula='MathScore ~ C(TestPrep) + NrSiblings + ParentMaritalStatus + ReadingScore', data=df)
res = mod.fit()
res.summary()


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              MathScore   R-squared:                       0.669
Model:                            OLS   Adj. R-squared:                  0.669
Method:                 Least Squares   F-statistic:                 1.474e+04
Date:                Sun, 07 Jan 2024   Prob (F-statistic):               0.00
Time:                        22:05:06   Log-Likelihood:            -1.0471e+05
No. Observations:               29117   AIC:                         2.094e+05
Df Residuals:                   29112   BIC:                         2.095e+05
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
=======================================================================================
                          coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------
Intercept               6.3552      0.302     21.030      0.000       5.763       6.947
C(TestPrep)[T.1]        1.1073      0.113      9.817      0.000       0.886       1.328
NrSiblings              0.0154      0.036      0.426      0.670      -0.056       0.086
ParentMaritalStatus    -0.0410      0.076     -0.536      0.592      -0.191       0.109
ReadingScore            0.8570      0.004    239.501      0.000       0.850       0.864
==============================================================================
Omnibus:                      233.653   Durbin-Watson:                   1.987
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              155.726
Skew:                          -0.010   Prob(JB):                     1.53e-34
Kurtosis:                       2.642   Cond. No.                         422.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Questions:

a) Using the summary, write down the formula with coefficients

$MathScore = 6.3522 + 1.1073*TestPrep + 0.0154*NrSiblings + (-0.0410)*ParentMaritalStatus + 0.8570*ReadingScore $

b) Are there any statisticaly important features?

Intercept, TestPrep and ReadingScore are statistically significant (P>|t|).

c) Let's interpret the data, if students do test preparation, how does that affect the outcome on the math exam? 

if a student does test preparation (TestPrep=1), the expected change in MathScore is an increase of approximately 1.1073 units. This suggests that, on average, students who undertake test preparation tend to score about 1.1073 points higher on the math exam compared to those who do not, after accounting for the other variables in the model.

d) Are students who practice sport better at math exams or worse? Using linear regression, model the MathScore using only one feature: PracticeSport. Don't include intercept. Using summary, decide which students do the best at math exams - those who never practice sport, sometimes practice sport or those who do it regularly?

In [26]:
# Declare the model
mod = smf.ols(formula='MathScore ~ C(PracticeSport) - 1 ', data=df)
res = mod.fit()
res.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              MathScore   R-squared:                       0.006
Model:                            OLS   Adj. R-squared:                  0.006
Method:                 Least Squares   F-statistic:                     87.35
Date:                Mon, 08 Jan 2024   Prob (F-statistic):           1.51e-38
Time:                        15:41:45   Log-Likelihood:            -1.2074e+05
No. Observations:               29117   AIC:                         2.415e+05
Df Residuals:                   29114   BIC:                         2.415e+05
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
=======================================================================================
                          coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------
C(PracticeSport)[0]    64.1576      0.244    262.426      0.000      63.678      64.637
C(PracticeSport)[1]    67.8441      0.150    452.998      0.000      67.551      68.138
C(PracticeSport)[2]    66.2896      0.126    526.520      0.000      66.043      66.536
==============================================================================
Omnibus:                      234.462   Durbin-Watson:                   1.997
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              214.722
Skew:                          -0.171   Prob(JB):                     2.36e-47
Kurtosis:                       2.756   Cond. No.                         1.94
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Students who practice sport have higher average math score. Students who practice sometimes tend to preform the best on math exams, followed by those who practice regularly, and students who do not practice sport have the lowest average mathscore. 